### PR TITLE
Rspec custom matcher: `on_event` with arguments

### PR DESCRIPTION
**Pull request by zacviandier**
- Give the possibility to pass arguments to the `on_event` function of the Rspec custom matcher.
- Add a example to the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -765,6 +765,8 @@ expect(job).to allow_event :run
 expect(job).to_not allow_event :clean
 expect(job).to allow_transition_to(:running)
 expect(job).to_not allow_transition_to(:cleaning)
+# on_event also accept arguments
+expect(job).to transition_from(:sleeping).to(:running).on_event(:run, :deframengation)
```
Comment (alto) [inline README.md]: Please fix the typo.

**Comment by alto** [comment]
@zacviandier Nice enhancement. Please fix the typo and I'm happy to merge it :+1:

**Comment by zacviandier** [comment]
Done ;)

**Comment by alto** [comment]
@zacviandier Lovely. Thanks a lot for your help!

**Comment by alto** [comment]
I just released version `4.8.0` including the change.

**Pull request merged**